torch/sparse/_semi_structured_conversions.py

```diff
@@ -1,7 +1,22 @@
 import torch
 
-@torch.compile
-def sparse_semi_structured_from_dense(dense):
+
+def _is_compile_supported(device_type):
+    import torch._dynamo
+
+    compile_supported = torch._dynamo.is_dynamo_supported()
+    if device_type == "cpu":
+        pass
+    elif device_type == "cuda" and compile_supported:
+        from torch._inductor.utils import has_triton
+
+        compile_supported = has_triton()
+    else:
+        compile_supported = False
+    return compile_supported
```

Comment:
Nope, find a good spot for this (maybe torch._dynamo.utils) and factor both uses of it to use this one.

Reply:
Sure, that was my intention - just wanted to see is it going to pass this failing test this way.

Reply:
Fixed now.